bundler: support binary/asset imports (e.g. PNG via import attributes)

## Problem

When a module uses Deno's raw-bytes import attribute to bundle an asset directly:

```ts
import myimageBytes from "./ads/myimage.png" with { type: "bytes" };
```

the `bundle` management command fails with:

```
✘ [ERROR] [unreachable] Not an ESM module. [plugin deno-loader]

    src/soris/arenas/soris_arena.ts:9:25:
      9 │ import myimageBytes from "./ads/myimage.png" with { type: "bytes" };
        ╵                          ~~~~~~~~~~~~~~~~~~~
```

The `@luca/esbuild-deno-loader` plugin intercepts the import, tries to validate the PNG as an ESM module, and rejects it because it contains no `export`/`import` statements.

## Root cause

`buildSWBundle` in `src/staticfiles/commands/bundle.ts` calls `esbuild.build()` without a `loader` map:

```ts
const result = await esbuild.build({
  entryPoints: [...],
  bundle: true,
  format: "esm",
  platform: "browser",
  plugins,   // only alexi-templates + denoPlugins — no asset loader
  // ← no `loader` option
});
```

esbuild's default loader for `.png` (and other binary formats) is `"file"` on `platform: "browser"`, but `@luca/esbuild-deno-loader` intercepts the resolution first and fails before esbuild's own loader logic runs.

## Proposed fix

Add a `loader` map to the `esbuild.build()` call that covers common binary/asset file extensions, mapping them to esbuild's `"binary"` loader (which emits the raw bytes as a `Uint8Array` — matching Deno's `with { type: "bytes" }` semantics):

```ts
const result = await esbuild.build({
  // ... existing options ...
  loader: {
    ".png":  "binary",
    ".jpg":  "binary",
    ".jpeg": "binary",
    ".gif":  "binary",
    ".webp": "binary",
    ".ico":  "binary",
    ".svg":  "binary",
    ".wasm": "binary",
    ".ttf":  "binary",
    ".otf":  "binary",
    ".woff": "binary",
    ".woff2": "binary",
  },
});
```

Alternatively, expose it as a `BuildSWBundleOptions` field (e.g. `loaders?: Record<string, string>`) so callers can extend it without modifying Alexi internals.

## Workaround (for now)

Use a runtime `fetch` instead of the import attribute:

```ts
// instead of: import bytes from "./ads/myimage.png" with { type: "bytes" };
const resp = await fetch(new URL("./ads/myimage.png", import.meta.url));
const bytes = new Uint8Array(await resp.arrayBuffer());
```

This avoids the bundler entirely but requires the file to be served as a static asset — defeating the purpose of bundling it inline.

## Environment

- `@alexi/staticfiles`: `0.48.4`
- `esbuild`: `0.24.2`
- `@luca/esbuild-deno-loader`: `0.11.1`
- Deno: `2.x` with `--unstable-raw-imports`

--- a/src/staticfiles/commands/bundle.ts
+++ b/src/staticfiles/commands/bundle.ts
@@ -454,6 +454,31 @@ export async function rewriteHtmlReferences(
 // =============================================================================
 
 /**
+ * Default binary asset loaders used by {@link buildSWBundle}.
+ *
+ * Maps common binary/asset file extensions to esbuild's `"binary"` loader,
+ * which emits the raw bytes as a `Uint8Array` — matching Deno's
+ * `with { type: "bytes" }` import-attribute semantics.
+ *
+ * Callers can override individual entries or add new ones via the
+ * {@link BuildSWBundleOptions.loaders} option.
+ */
+export const DEFAULT_ASSET_LOADERS: Record<string, string> = {
+  ".png": "binary",
+  ".jpg": "binary",
+  ".jpeg": "binary",
+  ".gif": "binary",
+  ".webp": "binary",
+  ".ico": "binary",
+  ".svg": "binary",
+  ".wasm": "binary",
+  ".ttf": "binary",
+  ".otf": "binary",
+  ".woff": "binary",
+  ".woff2": "binary",
+};
+
+/**
  * Options for {@link buildSWBundle}.
  */
 export interface BuildSWBundleOptions {
@@ -494,6 +519,21 @@ export interface BuildSWBundleOptions {
    * When `true`, `entryNames` is ignored and `[name]` is always used.
    */
   isServiceWorker?: boolean;
+  /**
+   * Additional or override esbuild loader mappings for binary/asset file
+   * extensions.
+   *
+   * These are merged on top of {@link DEFAULT_ASSET_LOADERS}, so you only need
+   * to supply the entries you want to add or change.  Setting an extension to
+   * `"file"` opts that type out of inline binary embedding.
+   *
+   * @example
+   * ```ts
+   * // Treat .mp4 as binary and opt .svg out of the default binary loader
+   * loaders: { ".mp4": "binary", ".svg": "file" }
+   * ```
+   */
+  loaders?: Record<string, string>;
 }
 
 /**
@@ -525,7 +565,6 @@ export async function buildSWBundle(
     minify = false,
     templates = [],
   } = options;
-
   const cwd = options.cwd ?? Deno.cwd();
 
   // Lazy-import esbuild and esbuild-deno-loader to prevent these server-only
@@ -644,6 +683,15 @@ export async function buildSWBundle(
     effectiveEntryPoint = { in: virtualEntryIn, out: originalBasename };
   }
 
+  // Merge default binary asset loaders with any caller-supplied overrides.
+  // This ensures that extensions like .png, .wasm, etc. are handled by
+  // esbuild's "binary" loader (emitting raw bytes as Uint8Array) before
+  // @luca/esbuild-deno-loader has a chance to reject them as non-ESM modules.
+  const effectiveLoaders: Record<string, esbuild.Loader> = {
+    ...DEFAULT_ASSET_LOADERS,
+    ...(options.loaders ?? {}),
+  } as Record<string, esbuild.Loader>;
+
   const result = await esbuild.build({
     entryPoints: [effectiveEntryPoint] as
       | [string]
@@ -663,6 +711,7 @@ export async function buildSWBundle(
     external: [],
     treeShaking: true,
     keepNames: true,
+    loader: effectiveLoaders,
     // Bound node_modules resolution to the project root so esbuild does not
     // walk up to the filesystem root (e.g. C:\) on Windows.
     absWorkingDir: cwd,

--- a/src/staticfiles/commands/mod.ts
+++ b/src/staticfiles/commands/mod.ts
@@ -9,3 +9,9 @@
 export { BundleCommand } from "./bundle.ts";
 export { CollectStaticCommand } from "./collectstatic.ts";
 export { RunServerCommand } from "./runserver.ts";
+export { buildSWBundle, DEFAULT_ASSET_LOADERS } from "./bundle.ts";
+export type {
+  BuildSWBundleOptions,
+  DiscoveredTemplate,
+  StaticFilesManifest,
+} from "./bundle.ts";

